overlay/mobile-nixos/kernel/builder.nix

```diff
@@ -203,7 +206,14 @@ stdenv.mkDerivation (inputArgs // {
       sed -i scripts/ld-version.sh -e "s|/usr/bin/awk|${buildPackages.gawk}/bin/awk|"
     fi
   ''
-    + maybeString prePatch
+  + optionalString (outOfTreeDtbs != null) ''
+    # We use tar to preserve directory structure in the source files
+    # (e.g. qcom/foo.dtb shoudl end up in arch/arm/boot/dts/qcom/foo.dtb).
+    # The options are to ensure the files/dirs are created writable so that
+    # sed can edit them in postPatch
+    tar -C ${outOfTreeDtbs} -cf - . | tar -C arch/arm/boot/dts/ --mode=a+w --no-overwrite-dir -xvvpf -
```

Comment:
Aren't those `outOfTreeDts`es?

We probably want to copy the structure at the root. AArch64 dts would be at `arch/arm64/boot/dts/qcom`... and then there's non-qualcomm device trees.

If instead the out of tree DTSes were rooted at `arch/arm/boot/dts/qcom/*.dts` that would make it work in a more universal manner.

Reply:
> AArch64 dts would be at `arch/arm64/boot/dts/qcom`

Hmm. This *is* an aarch64 device, but the kernel build expects them in `arm` not `arm64`. Maybe that's unique to this kernel lineage and not a generic thing.

I think you're right to be unsure,  it feels like there are too many variables involved to make this generic - given, as you say, that a patch is needed anyway

Reply:
Exactly, it's because it's an AArch64 with an ARM soul :).

Though *some* actually-AArch64 devices *do* put DTS files in `arch/arm/boot/dts`, which makes this **even more confusing**!!

Reply:
64c1ae4 fixes this